LibreNMS/Device/YamlDiscovery.php

```diff
@@ -111,7 +111,13 @@ public static function replaceValues($name, $index, $count, $data, $pre_cache)
         $value = dynamic_discovery_get_value($name, $index, $data, $pre_cache);
         if (is_null($value)) {
             // built in replacements
+            // prepare the $subindexX match variable replacement
+            $subindexes = explode('.', $index);
+            foreach (array_keys($subindexes) as $pos) {
```

Comment:
You didn't initialize $subindexStr, you could probably make this a little more readable.

Reply:
done